plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3select/TestS3SelectPushdown.java

```diff
@@ -13,34 +13,204 @@
+import io.trino.plugin.hive.metastore.Partition;
+import io.trino.plugin.hive.metastore.Storage;
+import io.trino.plugin.hive.metastore.StorageFormat;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.testing.TestingConnectorSession;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe;
 import org.apache.hadoop.mapred.JobConf;
 import org.apache.hadoop.mapred.TextInputFormat;
+import org.apache.hive.hcatalog.data.JsonSerDe;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Properties;
+
 import static io.trino.hadoop.ConfigurationInstantiator.newEmptyConfiguration;
+import static io.trino.plugin.hive.HiveMetadata.SKIP_FOOTER_COUNT_KEY;
+import static io.trino.plugin.hive.HiveMetadata.SKIP_HEADER_COUNT_KEY;
+import static io.trino.plugin.hive.HiveType.HIVE_BINARY;
+import static io.trino.plugin.hive.s3select.S3SelectPushdown.isCompressionCodecSupported;
+import static io.trino.plugin.hive.s3select.S3SelectPushdown.isSplittable;
+import static io.trino.plugin.hive.s3select.S3SelectPushdown.shouldEnablePushdownForTable;
+import static io.trino.spi.session.PropertyMetadata.booleanProperty;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_LIB;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
 
 public class TestS3SelectPushdown
 {
     private TextInputFormat inputFormat;
+    private ConnectorSession session;
+    private Table table;
+    private Partition partition;
+    private Storage storage;
+    private Column column;
+    private Properties schema;
+    private static final String S3_SELECT_PUSHDOWN_ENABLED = "s3_select_pushdown_enabled";
 
     @BeforeClass
     public void setUp()
     {
         inputFormat = new TextInputFormat();
         inputFormat.configure(new JobConf(newEmptyConfiguration()));
+
+        session = TestingConnectorSession.builder()
+                .setPropertyMetadata(List.of(booleanProperty(
+                        S3_SELECT_PUSHDOWN_ENABLED,
+                        "S3 Select pushdown enabled",
+                        true,
+                        false)))
+                .setPropertyValues(Map.of(S3_SELECT_PUSHDOWN_ENABLED, true))
+                .build();
+
+        column = new Column("column", HiveType.HIVE_BOOLEAN, Optional.empty());
+
+        storage = Storage.builder()
+                .setStorageFormat(StorageFormat.fromHiveStorageFormat(HiveStorageFormat.TEXTFILE))
+                .setLocation("location")
+                .build();
+
+        partition = new Partition("db", "table", emptyList(),
```

Comment:
nit: Through the project we prefer to stick with either all arguments on the same line, or each argument on a new line. e.g.:
```
partition = new Partition(
  "db",
  ''table",
  ...
```

(here and in other places through this class)